### DISCUSSION TITLE
User ID to Username#Discrim

### DISCUSSION THREAD

**Discussion by Atharvnadkarni** [discordjs/discord.js#5699] [Q&A] [answered]
I am trying to make a discord bot with discord.js, but I have a problem with the ban and kick commands.

When a person says `!ban @athroid`, I want the bot to respond with "athroid#9322 banned successfully". I have a `MemberTarget` variable with the user ID, and I want to convert it into username and tag (like 792330397307895818  to athroid#9322).

Here is my code:

   ```js
 if (message.member.roles.cache.some(r => (r.name === "Moderator" || r.name === "Administrator"))) {
                const member = message.mentions.users.first();
                if (member) {
                    const MemberTarget = message.guild.members.cache.get(member.id);
                    const MemberString = MemberTarget.toString();
                    MemberTarget.kick();
                    message.channel.send(`${MemberString.tag} has been kicked successfully.`).catch(error => console.error(`Error occured; ${MemberString.tag} was not kicked.`));
                } else {
                    message.channel.send("Sorry, you did not mention a valid user to kick.");
                }
            } else {
                message.channel.send("Um.. it looks like you don't have the right permissions. In order to get the permissions, DM your server owner to kick the user.");
           }
```

But this happens when I run the bot:

![](https://preview.redd.it/duzbyzk840271.png?width=398&format=png&auto=webp&s=4306e9c81c3ddc22354b5a89f2d1ab933ceddd00)

It should say AtharvNadkarni#4244, not "undefined".

But nicely enough, my alt is removed from the server:

![](https://preview.redd.it/dm0xtwej40271.png?width=220&format=png&auto=webp&s=ef82193d83a4d3a7024c3d30dbfbaaa2f1d1839c)

But why does it say "undefined"?

If I could, I would post the question on StackOverflow. But sadly, I cannot do it for 2 days because of my recent downvotes. Can you help me?

I need text in my question, so don't say I have to go on Gist.

**Comment by kyranet** [comment, marked as answer]
> But why does it say "undefined"?

Because `MemberString.tag` is similar to doing `'81440962496172032'.tag`, and the `tag` property does not exist in strings.

What you want to do is [`MemberTarget`](https://discord.js.org/#/docs/main/stable/class/GuildMember)`.`[`user`](https://discord.js.org/#/docs/main/stable/class/GuildMember?scrollTo=user)`.`[`tag`](https://discord.js.org/#/docs/main/stable/class/User?scrollTo=tag).
